src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/Codes/Codes.tsx

```diff
@@ -119,23 +150,59 @@ const Codes = ({
     setSavedCode(cqlCode);
   };
 
+  // Checks if new code is a duplicate based on chip ids.
   const addNewCode = () => {
     if (selectedCodeSystemName === "Custom") {
-      const customCqlCode = new CQL.Code(
-        customCodeConcept,
-        customCodeSystemName, // What is the oid for custom CS
-        null,
-        customCodeConcept
-      );
-      handleChange(customCqlCode);
+      const newCodeId = `${customCodeSystemName}-${customCodeConcept}`;
+      const existingCode = _.filter(chips, _.matches({ id: newCodeId }));
+      if (_.isEmpty(existingCode)) {
+        const customCqlCode = new CQL.Code(
+          customCodeConcept,
+          customCodeSystemName, // What is the oid for custom CS
+          null,
+          customCodeConcept
+        );
+        handleChange(customCqlCode);
+        setSelectedCodeSystemName("");
+        setCustomCodeSystemName("");
+        setCustomCodeConcept("");
+      }
     } else {
-      handleChange(savedCode);
+      const newCodeId = `${codeSystems[savedCode.system]}-${savedCode.code}`;
+      const existingCode = _.filter(chips, _.matches({ id: newCodeId }));
+      if (_.isEmpty(existingCode)) {
+        handleChange(savedCode);
+        setSelectedCodeSystemName("");
+        setSelectedCodeConcept(null);
+      }
     }
   };
 
+  // Deletes the chip & updates JSON
+  const handleDeleteCode = (chip) => {
+    const remainingChips = chips.filter((c) => {
+      return c.id !== chip.id;
+    });
+    const codeId = chip.id.split("-")[1];
```

Comment:
this logic would fail for the codes with hyphens in them e.g. codesystem= SNOMED & code= 1004-1, id=SNOMED-1004-1

Reply:
I didn't know that was an option, interesting

Reply:
Not ideal but I used underscore, Although that got me thinking, if a user adds "_" in the string of Code for Custom Code system, the delete doesn't work, but thats an edge case and I am not sure if there is a better way to uniquely identify a particular code.